src/modals/SettingsModal/index.tsx

```diff
@@ -16,50 +16,190 @@ You should have received a copy of the GNU Lesser General Public License
+  { value: 'wallets', label: i18next.t('Wallets') },
+  { value: 'network', label: i18next.t('Network') },
+  { value: 'devtools', label: i18next.t('Developer tools') }
+]
+
 const SettingsModal = ({ onClose }: SettingsModalProps) => {
   const { t } = useTranslation()
-  const tabs = [
-    { value: 'general', label: t('General'), component: <GeneralSettingsSection /> },
-    { value: 'client', label: t('Networks'), component: <NetworkSettingsSection /> },
-    { value: 'wallets', label: t('Wallets'), component: <WalletsSettingsSection /> }
-  ]
-  const [currentTab, setCurrentTab] = useState<TabItem>(tabs[0])
+  const theme = useTheme()
 
-  const tabsI18n = tabs.map((tab) => ({ ...tab, label: tab.label }))
+  const [currentTab, setCurrentTab] = useState<TabItem>(tabs[0])
 
   return (
-    <CenteredModal title={t`Settings`} onClose={onClose}>
-      <PanelContentContainer>
-        <TabBar items={tabsI18n} onTabChange={(tab) => setCurrentTab(tab)} activeTab={currentTab} />
-        {tabsI18n.find((t) => t.value === currentTab.value)?.component}
-        <Section>
-          <VersionNumber>
-            {t`Version`}: {import.meta.env.VITE_VERSION}
-          </VersionNumber>
-        </Section>
-      </PanelContentContainer>
-    </CenteredModal>
+    <ModalContainer onClose={onClose}>
+      <CenteredBox role="dialog" {...fadeInOutBottomFast}>
+        <TabTitlesColumn>
+          <TabTitlesColumnHeader>
+            <ColumnTitle>
+              <Settings color={theme.font.secondary} strokeWidth={1} />
+              {t('Settings')}
+            </ColumnTitle>
+          </TabTitlesColumnHeader>
+          <TabTitlesColumnContent>
+            <TabTitles>
+              {tabs.map((tab) => (
+                <TabTitleButton
+                  key={tab.value}
+                  role="secondary"
+                  wide
+                  transparent={currentTab.value !== tab.value}
+                  borderless={currentTab.value !== tab.value}
+                  onClick={() => setCurrentTab(tab)}
+                >
+                  {tab.label}
+                </TabTitleButton>
+              ))}
+            </TabTitles>
+            <Version>
+              {t('Version')}: {import.meta.env.VITE_VERSION}
+            </Version>
+          </TabTitlesColumnContent>
+        </TabTitlesColumn>
+        <TabContentsColumn>
+          <ColumnHeader>
+            <ColumnTitle>{currentTab.label}</ColumnTitle>
+            <CloseButton aria-label={t`Close`} onClick={onClose}>
+              <X />
+            </CloseButton>
+          </ColumnHeader>
+          <Scrollbar translateContentSizeYToHolder>
+            <ColumnContent>
+              {
+                {
+                  general: <GeneralSettingsSection />,
+                  wallets: <WalletsSettingsSection />,
+                  network: <NetworkSettingsSection />,
+                  devtools: <DevToolsSettingsSection />
+                }[currentTab.value]
```

Comment:
NIT: Same remark as before, I would extract this.

Reply:
Same reply as above 😇